### PR TITLE
1pif

### PR DESCRIPTION
**Pull request by perry-mitchell**
Add 1password import support

### CODE REVIEW

File: source/importers/1PasswordImporter.js [hunk 1]
```diff
@@ -1,9 +1,93 @@
...
-	};
+const {
+	convert1pifToJSON
+} = require("../tools/1password.js");
 
-})(module);
+const {
+	Archive
+} = Buttercup;
+const readFile = pify(fs.readFile);
```
Comment (sallar) [inline source/importers/1PasswordImporter.js]: [`fs-extra`](https://github.com/jprichardson/node-fs-extra) returns Promises too. but this is good as well. just saying

File: source/importers/KeePass2XMLImporter.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+"use strict";
```
Comment (sallar) [inline source/importers/KeePass2XMLImporter.js]: Is this file just been renamed?

File: tests/cli/1PasswordImportTests.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		},
+
+        containsGroups: function(test) {
+			ButtercupImporter
+				.importFrom1PIF(this.exampleArchive)
+				.then(function(archive) {
+					const rootGroup = archive.getGroups()[0];
+					test.ok(/Imported 1password archive/i.test(rootGroup.getTitle()), "Root group should be named correctly");
+					const generalGroup = rootGroup.findGroupsByTitle("General")[0];
+                    test.ok(generalGroup instanceof Group, "General group should exist");
```
Comment (sallar) [inline tests/cli/1PasswordImportTests.js]: There is one extra space here 🤔

File: tests/cli/1PasswordImportTests.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                    const subGroup = generalGroup.findGroupsByTitle("Sub")[0];
+                    test.ok(subGroup instanceof Group, "Sub group should exist");
+				})
+                .then(test.done);
+		},
+
+        containsEntries: function(test) {
```
Comment (sallar) [inline tests/cli/1PasswordImportTests.js]: and here too. I think if you cut the lines and paste them again vscode will fix it all

image

File: tests/cli/1PasswordImportTests.js (hunk 3 above)
Comment (sallar) [inline tests/cli/1PasswordImportTests.js]: Oh I figured it out, the other lines are using tabs

File: source/tools/1passwordEntry.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                entry.password = field.value;
+            }
+        });
+    } else if (ENTRY_CREDITCARD.test(rawItem.typeName)) {
+        entry.username = rawItem.secureContents.ccnum;
+        entry.password = rawItem.secureContents.cvv;
+        entry.meta.type = rawItem.secureContents.type;
+        entry.meta.cardholder = rawItem.secureContents.cardholder;
+        entry.meta["expiry-year"] = rawItem.secureContents.expiry_yy;
+        entry.meta["expiry_month"] = rawItem.secureContents.expiry_mm;
```
Comment (sallar) [inline source/tools/1passwordEntry.js]: `expiry_month` should probably be `expiry-month` or the other way around

File: source/importers/1PasswordImporter.js (hunk 1 above)
Comment (perry-mitchell) [inline source/importers/1PasswordImporter.js]: I also like fs-extra, but for one function it's a bit easier and simpler imo to just promisify it..

File: source/importers/KeePass2XMLImporter.js (hunk 2 above)
Comment (perry-mitchell) [inline source/importers/KeePass2XMLImporter.js]: Yes.. The old name didn't make any sense.

File: tests/cli/1PasswordImportTests.js (hunk 3 above)
Comment (perry-mitchell) [inline tests/cli/1PasswordImportTests.js]: Ah.. I'll normalise those then :)

File: tests/cli/1PasswordImportTests.js (hunk 4 above)
Comment (perry-mitchell) [inline tests/cli/1PasswordImportTests.js]: Will do!

File: source/tools/1passwordEntry.js (hunk 5 above)
Comment (perry-mitchell) [inline source/tools/1passwordEntry.js]: It should be `expiry-month`.. Good catch!

**Pull request merged**